### PR TITLE
Clone JSpecify when needed and add test dependency

### PR DESCRIPTION
**Pull request by Ao-senXiong**
Add cloneAndBuildJspecify task and make test depends on it.

This make ./gradlew test will success.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -104,6 +105,43 @@ switch (JavaVersion.current()) {
         break;
 }
 
+task cloneAndBuildJspecify {
+    description 'Obtain or update JSpecify.'
+    doLast {
+        if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
+            if (file(jspecifyHome).exists()) {
```
Comment (wmdietl) [inline build.gradle:112]: Can't you use the `createCloneTask` to do this?

File: build.gradle [hunk 2]
```diff
@@ -104,6 +105,43 @@ switch (JavaVersion.current()) {
         break;
 }
 
+task cloneAndBuildJspecify {
+    description 'Obtain or update JSpecify.'
+    doLast {
+        if (JavaVersion.current() > JavaVersion.VERSION_1_8) {
```
Comment (wmdietl) [inline build.gradle:111]: It's clearer if you exit early if the condition isn't met. Add a comment why.

File: build.gradle [hunk 3]
```diff
@@ -104,6 +105,43 @@ switch (JavaVersion.current()) {
...
+            exec {
+                println "Executing in directory: ${jspecifyHome}"
+                workingDir jspecifyHome
+                commandLine 'sh', '-c', './gradlew assemble'
+            }
+        }
+    }
+}
+
+test.dependsOn cloneAndBuildJspecify
```
Comment (wmdietl) [inline build.gradle:143]: Not all tests require JSpecify. Can we target this only to the JSpecify tests?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks for looking into this!

File: build.gradle (hunk 1 above)
Comment (Ao-senXiong) [reply build.gradle]: ok, used createCloneTask to simplify the task.

File: build.gradle (hunk 2 above)
Comment (Ao-senXiong) [reply build.gradle]: already add the comment.

File: build.gradle (hunk 3 above)
Comment (Ao-senXiong) [reply build.gradle]: It seems like even if add dependency to NullnessNullMarkedTest, it did not automatically  apply to the Test task

File: build.gradle [hunk 4]
```diff
@@ -88,6 +88,7 @@ ext {
         plumeUtil       : '1.8.1',
         reflectionUtil  : '1.1.0',
     ]
+    jspecifyHome = '../jspecify'
```
[the remark this answers is not in the document]
Comment (Ao-senXiong) [reply build.gradle]: Done

**Review by wmdietl — APPROVED** [review]
Thanks for the updates, looks good!

**Pull request merged**